api/httpverbs.go

```diff
@@ -0,0 +1,86 @@
+package api
```

Comment:
This module (as well as `httpverbswithjson` feels like something someone must have written once or twice before.  
Can we use something like https://github.com/go-resty/resty instead please?